Sources/SQLiteNIO/SQLiteConnection+Hooks.swift

```diff
@@ -0,0 +1,1098 @@
+    /// Drop virtual table.
+    public static let dropVTable = SQLiteAuthorizerAction(rawValue: SQLITE_DROP_VTABLE)
+    /// Function call.
+    public static let function = SQLiteAuthorizerAction(rawValue: SQLITE_FUNCTION)
+    /// Savepoint operation.
+    public static let savepoint = SQLiteAuthorizerAction(rawValue: SQLITE_SAVEPOINT)
+    /// Copy operation.
+    public static let copy = SQLiteAuthorizerAction(rawValue: SQLITE_COPY)
+    /// Recursive operation.
+    public static let recursive = SQLiteAuthorizerAction(rawValue: SQLITE_RECURSIVE)
+}
+
+/// The response from a commit hook callback.
+public enum SQLiteCommitResponse: Sendable, Hashable {
+    /// Allow the commit to proceed.
+    case allow
+    /// Deny the commit (causes transaction rollback).
+    case deny
+}
+
+/// The response from an authorizer callback.
+public enum SQLiteAuthorizerResponse: Int32, Sendable, Hashable {
+    /// Allow the operation.
+    case allow = 0 // SQLITE_OK
+    /// Deny the operation.
+    case deny = 1 // SQLITE_DENY
+    /// Ignore the operation (treat column as NULL).
+    case ignore = 2 // SQLITE_IGNORE
+}
+
+/// Event produced by the authorizer hook.
+///
+/// Contains information about a database access attempt that requires authorization.
+public struct SQLiteAuthorizerEvent: Sendable, Hashable {
+    /// The type of database access being attempted.
+    public let action: SQLiteAuthorizerAction
+    /// The first parameter (meaning depends on action).
+    public let parameter1: String?
+    /// The second parameter (meaning depends on action).
+    public let parameter2: String?
+    /// The database name.
+    public let database: String?
+    /// The trigger or view name that caused the access.
+    public let trigger: String?
+}
+
+/// Event produced by the commit hook.
+///
+/// Contains information about a transaction commit that is about to occur.
+public struct SQLiteCommitEvent: Sendable, Hashable {
+    /// Timestamp (in the connection's wall-clock) when the commit was about to occur.
+    public let date: Date = Date()
+}
+
+/// Event produced by the rollback hook.
+///
+/// Contains information about a transaction rollback that has occurred.
+public struct SQLiteRollbackEvent: Sendable, Hashable {
+    /// Timestamp (in the connection's wall-clock) when the rollback happened.
+    public let date: Date = Date()
+}
+
+// MARK: - Hook Tokens and Identifiers
+
+extension SQLiteConnection {
+    /// Represents the different types of database hooks available.
+    public enum HookKind: Sendable, Hashable {
+        /// Update hook (fired on INSERT, UPDATE, DELETE operations)
+        case update
+        /// Commit hook (fired before transaction commits)
+        case commit
+        /// Rollback hook (fired when transaction rolls back)
+        case rollback
+        /// Authorizer hook (fired during statement preparation for access control)
+        case authorizer
+    }
```

Comment:
Why embed this particular enum in `SQLiteConnection` instead of making it global?